tests/Integration/Behaviour/Features/Context/Domain/Product/StockAssertionFeatureContext.php

```diff
@@ -263,20 +267,27 @@ public function assertInvalidOutOfStockType(): void
     /**
      * @param StockMovement[] $stockMovements
      * @param TableNode $table
+     * @param int $shopId
      */
-    private function assertStockMovementHistories(array $stockMovements, TableNode $table): void
+    private function assertStockMovementHistories(array $stockMovements, TableNode $table, int $shopId): void
     {
         $movementsData = $table->getColumnsHash();
 
-        Assert::assertEquals(count($movementsData), count($stockMovements));
+        Assert::assertEquals(count($movementsData), count($stockMovements), sprintf(
+            'Unexpected number of movements for shop %d expected %d but got %d instead',
+            $shopId,
+            count($movementsData),
+            count($stockMovements)
+        ));
         $index = 0;
         foreach ($movementsData as $movementDatum) {
             $stockMovement = $stockMovements[$index];
             Assert::assertEquals(
                 $movementDatum['employee'],
                 $stockMovement->getEmployeeName(),
                 sprintf(
-                    'Invalid employee name of stock movement event, expected "%s" instead of "%s"',
+                    'Invalid employee name of stock movement event for shop%d, expected "%s" instead of "%s"',
```

Comment:
```suggestion
                    'Invalid employee name of stock movement event for shop %d, expected "%s" instead of "%s"',
```

Reply:
unless you made this intentionally? But it might be confusing with the shop reference this way. Because we use lots of shops references called "shop1", "shop2".
Check other parts, this one is repeated in some places bellow :arrow_down:

Reply:
No it was a typo, thanks